File: cpp/program26.cpp
// program30.c

#include <iostream>
using namespace std;

class Number
{
    public:
        int iNo;

    Number(int X)
    {
        iNo = X;
    }

    bool CheckPrime()
    {
        int iCnt = 0;
        bool bFlag = true;

        if (iNo < 0)
        {
            iNo = -iNo;
        }

        for(iCnt = 2; iCnt <= (iNo/2); iCnt++)
        {
            if((iNo % iCnt) == 0)
            {
                bFlag = false;
                break;
            }
        }
        return bFlag;
    }

};

int main()
{
    int iValue = 0;
    bool bRet = false;

    cout<<"Enter Number"<<"\n";
    cin>>iValue;

    Number nobj(iValue);
    bRet = nobj.CheckPrime();

    if(bRet == true)
    {
        cout<<iValue<<" is Prime number"<<"\n";
    }
    else
    {
        cout<<iValue<<" is NOT Prime number"<<"\n";
    }

    return 0;
}

File: java/program26.java
import java.util.*;

class Number 
{
    public int iNo;

    public Number(int no)
    {
        iNo = no;
    }

    public boolean CheckPrime()
    {
        int iCnt = 0;
        boolean flag = true;

        if(iNo < 0)
        {
            iNo = -iNo;
        }

        for(iCnt = 2; iCnt <= (iNo/2); iCnt++)
        {
            if(iNo % iCnt == 0)
            {
                flag = false;
                break;
            }
        }
        return flag;
    }
}

class program26
{
    public static void main(String arg[])
    {
        int iValue = 0;
        boolean bRet = false;

        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter number");
        iValue = sobj.nextInt();

        Number obj = new Number(iValue);
        bRet = obj.CheckPrime();

        if(bRet == true)
        {
            System.out.println(iValue+" is prime number");
        }
        else
        {
            System.out.println(iValue+" is NOT prime number");
        }
    }
}